### PR TITLE
qemu: Add bound checks for memory assignment

### PR DESCRIPTION
**Pull request by simar7**
This PR add bounds checks for upper [[1]](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Virtualization_Administration_Guide/sect-mem-dump-off.html)  and lower possible memory
assignments to a virtual machine using the QEMU driver.

Signed-off-by: Simarpreet Singh <simar@linux.com>

### CODE REVIEW

**Comment by schmichael** [comment]
I'm curious what the use case for this PR is. [`structs.go`](https://github.com/hashicorp/nomad/blob/master/nomad/structs/structs.go#L1271) prevents sub-10mb memory allocations, and the scheduler won't alloc tasks to allocate more memory than the node has available.

**Comment by simar7** [comment]
@schmichael ah gotcha -- didn't see that. Do you think it's still useful to have the tests that are included here for catching those cases?

**Comment by schmichael** [comment]
I'd prefer not to as once we stop trusting the invariants guaranteed by the server we add a lot of code to all of our drivers. The tests only work because they bypass the validation code.

Have you had issues where the qemu driver improperly allocated memory?

**Comment by schmichael** [comment]
Ah! Qemu's limit is 4tb! Your PR makes a lot more sense now. Sorry for the confusion.

Simply change the lower bound check to 128 MB and I'll merge it! Also please run `gofmt -s` on `qemu_test.go` to make our linters happy.

Thanks and sorry again for my confusion!

**Comment by simar7** [comment]
@schmichael not a problem! I've run it through ```gofmt -s -w``` and it should be good to go!

File: client/driver/qemu.go [hunk 1]
```diff
@@ -170,7 +170,10 @@ func (d *QemuDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse
 	if d.driverConfig.Accelerator != "" {
 		accelerator = d.driverConfig.Accelerator
 	}
-	// TODO: Check a lower bounds, e.g. the default 128 of Qemu
+
+	if task.Resources.MemoryMB <= 0 || task.Resources.MemoryMB > 4000000 {
```
Comment (schmichael) [inline client/driver/qemu.go]: Should be <128 right?

Comment (simar7) [reply client/driver/qemu.go]: Right, sorry overlooked that. I've patched it in the latest commit.

**Comment by schmichael** [comment]
Looks great! Thanks a lot @simar7!

**Pull request merged**